### DISCUSSION TITLE
To boot SATA on Orange Pi 5

### DISCUSSION THREAD

**Discussion by hakotani-o** [Joshua-Riek/ubuntu-rockchip#501] [Q&A]
Thank you for your great work .

I have a question.
To boot SATA on Orange Pi 5,
Do I need to change the description to overlays=orangepi-5-sata in the 5.10.160-23 kernel?

Sorry for the poor English.
Thank you.

**Comment by Joshua-Riek** [comment]
Yes, and please update the bootloader with `sudo u-boot-install-sata-mtd /dev/mtdblock0`

**Reply by hakotani-o** [reply]
Thank you for your reply.
I update the bootloader.
Thank you for your advice.

**Reply by revolverve** [reply]
Still work in 24.04 !
Great work ,
big Thank you for supporting opi5 , knowing OrangePi lack of it..and very few users.

**Reply by hakotani-o** [reply]
Hi @revolverve.

On 24.04 U_BOOT_FDT_OVERLAYS="device-tree/rockchip/overlay/orangepi-5-sata.dtbo" add to /etc/default/u-boot.
Please check wiki -> Ubuntu 24.04 LTS -> Using a Device Tree Overlay.

Thank you.

**Reply by revolverve** [reply]
Hi @hakotani-o
Hmm , i just realise , i did not even had to one overlay thing ?
Booted sdcard , setup and updated everything , did @Joshua-Riek  sudo u-boot-install-sata-mtd /dev/mtdblock0 ,
clone sdcard to sata m.2 with gparted , within another distro, and pop the m.2 in ,Opi5 reboot from sata directly !!
Lately Armbian sata boot did not worked anymore  .
Not an usual full Ubuntu user , mainly derivative /compatible no systemd distro of it on pc , puppy linux and last 4 years Easyos ,but  ,do enjoy 24.04 on arm64  , it is big ,with plenty of  update  ,so  i guess ,a pretty secure os  with  more then i ever need in packages choices !!
 on Opi5 ,fast enought for a full bloated distro !  
  Indeed an interesting experience .
Thank you all .

**Reply by hakotani-o** [reply]
Hi @revolverve.

I use M.2 NGFF SATA To 7 Pin SATA3 Card 7Pin SATA3.0 SSD Adapter Converter Card For NGFF 2242 2260 2280 SSD.
![Screenshot 2024-10-12 at 13-58-54 M 2 ngff sata to 7ピンsata3カード 7ピン sata 3 0 ssdアダプター コンバーターカード ngff 2242 2260 2280 - AliExpress 7](https://github.com/user-attachments/assets/90cfe173-0600-4ef2-aa4b-9ca599fe9c17)

This kit allows you to easily swap disks and enjoy ubuntu-rockchip or armbian.

**Comment by ageevp** [comment]
But this rewrite this UEFI bootloader MTD

https://github.com/edk2-porting/edk2-rk3588/releases

which is unacceptable!!!!!

**Reply by hakotani-o** [reply]
On opi5, u-boot can boot from M.2 SATA SSD.

Boot from sd-card writing ubuntu-24.04-preinstalled-desktop-arm64-orangepi-5.img.xz.
Write rkspi_loader_sata.img to SPI flash using u-boot-install-sata-mtd.
Add fdt /lib/firmware/6.1.0-1025-rockchip/device-tree/rockchip/rk3588s-orangepi-5-sata.dtb to /boot/extconf/extlinux.conf.
reboot.
After boot, ls /dev/sd*.
You can see SSD disk.( For example sda )
sudo ubuntu-rockchip-install /dev/sda. ( copy from sd-card to SSD )
shutdown.
Remove the sd-card.
Power on.
Boot from SSD.

Thank you.
